include/alp/algorithms/cholesky.hpp

```diff
@@ -157,9 +165,364 @@ namespace alp {
+		RC cholesky_uptr_blk(
+			MatL &L,
+			const MatH &H,
+			const size_t &bs,
+			const Ring &ring = Ring(),
+			const Minus &minus = Minus(),
+			const Divide &divide = Divide()
+		) {
+			(void) divide;
+			const Scalar< D > zero( ring.template getZero< D >() );
+
+			if (
+				( nrows( L ) != nrows( H ) ) ||
+				( ncols( L ) != ncols( H ) )
+			) {
+				std::cerr << "Incompatible sizes in cholesky_uptr_blk.\n";
+				return FAILED;
+			}
+
+			RC rc = SUCCESS;
+
+			const size_t n = nrows( L );
+
+			Matrix< D, typename MatH::structure > LL( n );
+			rc = rc ? rc : set( LL, H );
+#ifdef DEBUG
+			if( rc != SUCCESS ) {
+				std::cout << "set failed\n";
+				return rc;
+			}
+#endif
+
+			//nb: number of blocks of (max) size bz
+			size_t nb = n / bs ;
+			if( n % bs != 0 ){
+				nb = nb + 1 ;
+			}
+
+
+			for( size_t i = 0; i < nb ; ++i ) {
+				size_t a = std::min( i * bs, n );
+				size_t b = std::min( ( i + 1 ) * bs, n );
+				size_t c = n;
+
+				auto range1 = utils::range( a, b );
+				auto range2 = utils::range( b, c );
+
+				//A11=L[i*bs:(i+1)*bs,i*bs:(i+1)*bs]
+				auto A11 = get_view( LL, range1, range1 );
+
+				//A21=L[(i+1)*bs:,i*bs:(i+1)*bs]
+				// for complex we should conjugate A21
+				auto A21 = get_view< structures::General >( LL, range1, range2 );
+
+				//A11=cholesky(A11)
+				auto A11_out = get_view( L, range1, range1 );
+
+				rc = rc ? rc : cholesky_uptr( A11_out, A11, ring );
+#ifdef DEBUG
+				if( rc != SUCCESS ) {
+					std::cout << "cholesky_uptr failed\n";
+					return rc;
+				}
+#endif
+
+				auto A21_out = get_view< structures::General >(	L, range1, range2 );
+				auto A11_out_T = get_view< alp::view::transpose >( A11_out );
+
+				rc = rc ? rc : algorithms::forwardsubstitution(
+					A11_out_T,
+					A21_out,
+					A21,
+					ring
+				);
+#ifdef DEBUG
+				if( rc != SUCCESS ) {
+					std::cout << "Forwardsubstitution failed\n";
+					return rc;
+				}
+#endif
+
+				Matrix< D, typename MatH::structure > Reflector( ncols(A21_out) );
+				rc = rc ? rc : set( Reflector, zero );
+				rc = rc ? rc : mxm( Reflector, get_view< alp::view::transpose >( A21_out ), A21_out, ring );
+
+				auto A22 = get_view( LL, range2, range2 );
+				rc = rc ? rc : foldl( A22, Reflector, minus );
+			}
+
+			return rc;
+		}
+
+		// inplace non-blocked versions
+		template<
+			typename MatL,
+			typename D = typename MatL::value_type,
+			typename Ring = Semiring< operators::add< D >, operators::mul< D >, identities::zero, identities::one >,
+			typename Minus = operators::subtract< D >,
+			typename Divide = operators::divide< D >,
+			std::enable_if_t<
+				is_matrix< MatL >::value &&
+				structures::is_a< typename  MatL::structure, structures::Square >::value &&
+				is_semiring< Ring >::value &&
+				is_operator< Minus >::value &&
+				is_operator< Divide >::value
+			> * = nullptr
+		>
+		RC cholesky_uptr(
+			MatL &L,
+			const Ring &ring = Ring(),
+			const Minus &minus = Minus(),
+			const Divide &divide = Divide()
+		) {
```

Comment:
Consider adding a check for size compatibility like in other variants

Reply:
Only one Square matrix as input, nothing to check!